metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -506,17 +501,43 @@ def build_aggregated_measures(
             data_sources_for_measures[data_source_names[0]].append(measure_spec)
 
         for data_source, measures in data_sources_for_measures.items():
-            logger.info(f"Building aggregated measures for {data_source}. Measures: {measures}")
-            node = self._build_aggregated_measures_for_single_data_source(
-                measure_specs=tuple(measures),
-                queried_linkable_specs=queried_linkable_specs,
-                where_constraint=where_constraint,
-                time_range_constraint=time_range_constraint,
-                cumulative=cumulative,
-                cumulative_window=cumulative_window,
-                cumulative_grain_to_date=cumulative_grain_to_date,
-            )
-            output_nodes.append(node)
+            grouped_measures_by_additiveness = group_measure_specs_by_additiveness(measures)
+            additive_measure_specs = grouped_measures_by_additiveness.additive_measures
+            grouped_semi_additive_measures = grouped_measures_by_additiveness.grouped_semi_additive_measures
+
+            # Build output node for additive measures
+            if additive_measure_specs:
+                logger.info(
+                    f"Building aggregated measures for {data_source} with additive measures: {additive_measure_specs}"
+                )
+                output_nodes.append(
+                    self._build_aggregated_measures_for_single_data_source(
```

Comment:
In retrospect, `build_aggregated_measures_for_single_data_source` is better named to something like `build_aggregated_measures_from_measure_source_node` now. What do you think?